ActionLogger/src/pl/jtat/logger/model/EventData.java

```diff
@@ -0,0 +1,62 @@
+package pl.jtat.logger.model;
+
+import pl.jtat.logger.C;
+import pl.jtat.logger.util.DateFormatter;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+public class EventData {
+
+    private UUID eventId;
+    private EventType eventType;
+    private Date eventDate;
+    private Boolean eventResult;
+    private List<KeyValueData> additionalEventData;
+
+    public EventData(EventType eventType) {
+        this(eventType, null, null);
+    }
+
+    public EventData(EventType eventType, List<KeyValueData> additionalEventData) {
+        this(eventType, null, additionalEventData);
+    }
+
+    public EventData(EventType eventType, Boolean result, List<KeyValueData> additionalEventData) {
+        eventId = UUID.randomUUID();
+        eventDate = Calendar.getInstance().getTime();
+        this.eventType = eventType;
+        this.eventResult = result;
+        this.additionalEventData = additionalEventData;
+    }
+
+    private String getAdditionalEventDataString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(C.Strings.BRACKET_CURLY_LEFT_WITH_SPACE);
+        for(int i = 0; i < additionalEventData.size(); i++) {
```

Comment:
Tak jak pisałem wyżej docelowo będziemy zapisywać logi do bazy. Więc jeżeli już robić zapisywanie w formie stringa do pliku (a może być dodatkowo) to taka konwersja eventa do postaci tekstu powinna znaleźć się przynajmniej w przeznaczonym do tego LogWriterze, bo w implementacji LogWritera zapisującego do bazy danych nie będziemy potrzebowali tej konwersji

Reply:
Funkcja służy do zwracania dodatkowych informacji o akcji w postaci Stringa. Stringa budowanego na podstawie pola zawartego w tej klasie i może zostać wykorzystana do wyświetlania np na konsoli. Zastanowiłabym się nad jakimś Helperem/Converterem ale nie w tej chwili (ponieważ tak jak napisałam jest to ściśle związane z zawartością obiektu a i sama funkcja jest wywoływana w obrębie innej funkcji w tej samej klasie).